packages/sdk/src/evm/common/any-evm-utils.ts

```diff
@@ -494,17 +498,83 @@ export async function getDeploymentInfo(
   create2Factory?: string,
 ): Promise<DeploymentInfo> {
   const compilerMetadata = await fetchPreDeployMetadata(metadataUri, storage);
+  let finalInfraContracts: InfraContractType[] = [];
+  let pluginTransactions: PrecomputedDeploymentTransaction[] = [];
+  let customParams: any[] = [];
+
   if (!create2Factory) {
     create2Factory = await getCreate2FactoryAddress(provider);
   }
 
+  const isPluginRouter: boolean = isFeatureEnabled(
+    AbiSchema.parse(compilerMetadata.abi),
+    "PluginRouter",
+  );
+  if (isPluginRouter) {
+    const pluginNames = await getPluginNames(metadataUri, storage);
+    const pluginUris = await Promise.all(
+      pluginNames.map((name) => {
+        return fetchPublishedContractURI(name);
+      }),
+    );
+    const pluginMetadata = await Promise.all(
+      pluginUris.map((uri) => {
+        return fetchPreDeployMetadata(uri, storage);
+      }),
+    );
+    const deploymentParams = await Promise.all(
+      pluginMetadata.map((metadata) => {
+        return constructDeploymentParamsForDeployer(
+          metadata,
+          provider,
+          storage,
+          create2Factory as string,
```

Comment:
don't force cast, unless absolutely necessary - leads to errors that are really annoying to debug